simulators/mujoco_simulator/revolve2/simulators/mujoco_simulator/_custom_mujoco_viewer.py

```diff
@@ -0,0 +1,388 @@
+                )
+            case _:
+                print("Didnt reach anything with mode: " + self._viewer_mode.value)
+
+        """These are default overlays, only change if you know what you are doing."""
+        if self._render_every_frame:
+            self._add_overlay(topleft, "", "")
+        else:
+            self._add_overlay(
+                topleft,
+                "Run speed = %.3f x real time" % self._run_speed,
+                "[S]lower, [F]aster",
+            )
+        self._add_overlay(
+            topleft, "Ren[d]er every frame", "On" if self._render_every_frame else "Off"
+        )
+        self._add_overlay(
+            topleft,
+            "Switch camera (#cams = %d)" % (self.model.ncam + 1),
+            "[Tab] (camera ID = %d)" % self.cam.fixedcamid,
+        )
+
+        self._add_overlay(topleft, "Shad[O]ws", "On" if self._shadows else "Off")
+        self._add_overlay(
+            topleft, "T[r]ansparent", "On" if self._transparent else "Off"
+        )
+        self._add_overlay(topleft, "[W]ireframe", "On" if self._wire_frame else "Off")
+        self._add_overlay(
+            topleft,
+            "Con[V]ex Hull Rendering",
+            "On" if self._convex_hull_rendering else "Off",
+        )
+        if self._paused is not None:
+            if not self._paused:
+                self._add_overlay(topleft, "Stop", "[Space]")
+            else:
+                self._add_overlay(topleft, "Start", "[Space]")
+                self._add_overlay(
+                    topleft, "Advance simulation by one step", "[right arrow]"
+                )
+        self._add_overlay(
+            topleft,
+            "Toggle geomgroup visibility (0-5)",
+            ",".join(["On" if g else "Off" for g in self.vopt.geomgroup]),
+        )
+        self._add_overlay(
+            topleft, "Referenc[e] frames", mujoco.mjtFrame(self.vopt.frame).name
+        )
+        self._add_overlay(topleft, "[H]ide Menus", "")
+        if self._image_idx > 0:
+            fname = self._image_path % (self._image_idx - 1)
+            self._add_overlay(topleft, "Cap[t]ure frame", "Saved as %s" % fname)
+        else:
+            self._add_overlay(topleft, "Cap[t]ure frame", "")
+
+        self._add_overlay(bottomleft, "FPS", "%d%s" % (1 / self._time_per_render, ""))
+
+        if self._mujoco_version >= (3, 0, 0):
+            self._add_overlay(
+                bottomleft, "Max solver iters", str(max(self.data.solver_niter) + 1)
+            )
+        else:
+            self._add_overlay(
+                bottomleft, "Solver iterations", str(self.data.solver_iter + 1)
+            )
+
+        self._add_overlay(
+            bottomleft, "Step", str(round(self.data.time / self.model.opt.timestep))
+        )
+        self._add_overlay(bottomleft, "timestep", "%.5f" % self.model.opt.timestep)
+
+    def _key_callback(
+        self, window: Any, key: Any, scancode: Any, action: Any, mods: Any
+    ) -> None:
```

Comment:
same here, can you call the parent function so less copying is needed?

Reply:
yes i used the parent here since we only add functionality